WebDriverAgentLib/Utilities/FBXCodeCompatibility.m

```diff
@@ -43,3 +43,28 @@ + (instancetype)fb_applicationWithPID:(pid_t)processID
 
 @end
 
+static BOOL FBShouldUseFirstMatchSelector = NO;
+static dispatch_once_t onceFirstMatchToken;
+@implementation XCUIElementQuery (FBCompatibility)
+
+- (XCUIElement *)fb_firstMatch
+{
+  SEL firstMatchSelector = NSSelectorFromString(@"firstMatch");
```

Comment:
Possibly, you could simply add firstMatch selector in our private headers as class extension (`@interface XCUIEment ()`) and keep  `respondsToSelector` check. That would be even faster